pavelib/prereqs.py

```diff
@@ -132,10 +132,18 @@ def node_prereqs_installation():
     """
     Configures npm and installs Node prerequisites
     """
+    cb_error_text = "cb() never called"
     sh("test `npm config get registry` = \"{reg}\" || "
        "(echo setting registry; npm config set registry"
        " {reg})".format(reg=NPM_REGISTRY))
-    sh('npm install')
+
+    # Error handling around a race condition that produces "cb() never called" error. This
+    # ought to disappear when we upgrade npm to 3 or higher. TODO: clean this up when we do that.
+    capture = sh('npm install', capture=True, ignore_error=True)
```

Comment:
@jzoldak Thanks for the feedback. Unfortunately for `capture` (the var) to work, we also need to have `ignore_error`. Otherwise it'll capture the `cb() never called` and exit immediately.

Indeed I looked at unit testing this and even our base `PaverTestCase` won't apply here, because we need to mock the sh call itself due to our use of `capture`. 

That said, you are right; if there is a *different* kind of error, it'll just move onto the next step rather than failing. So I'll need to build a solution for that case.